psi4/src/psi4/libfock/cubature.cc

```diff
@@ -3632,7 +3649,11 @@ void MolecularGrid::buildGridFromOptions(MolecularGridOptions const& opt,
     radial_grids_.clear();
     spherical_grids_.clear();
 
+    radial_grids_.resize(molecule_->natom());
+    spherical_grids_.resize(molecule_->natom());
```

Comment:
Any need to handle `opt.namedGrid` differently here?

Reply:
It looks like opt.namedGrid can be:

- -1 (Default) if not using an explicitly named grid (uses radial_grids_ and spherical_grids_)
- 0 to use the SG0 grid (doesn't touch radial_grids_ or spherical_grids_)
- 1 to use the SG1 grid (doesn't touch radial_grids_ or spherical_grids_)

I think that in this overloaded version of buildGridFromOptions(), the fact that parameters passed are passed in (rs, ws, Ls) precludes the grid from being SG0 or SG1. I'll admit that I'm learning the details of DFT alongside looking at this code so I could be wrong about how this implementation works.

Reply:
SG2 and SG3 aren't implemented in Psi4 yet?

Reply:
It looks that way to me:

https://github.com/psi4/psi4/blob/f687f496d84aeb72162db36d7da4b5577ed7a4cb/psi4/src/psi4/libfock/cubature.cc#L2465-L2474

Reply:
@zachglick sounds plausible, thanks. @susilehtola, actually none of the SG grids are in Psi4; SG0 and 1 are just placeholders.